trestlebot/tasks/sync_cac_catalog_task.py

```diff
@@ -0,0 +1,241 @@
+                            if not control.controls:
+                                control.controls = []
+                            parent = control
+                            found = True
+                            break
+                    if not found:
+                        raise ValueError("Nested control path before parent control definition")
+            # 4. Find the associated oscal control to the cac control
+            # 4a. Map the cac control onto a new oscal control
+            new_control = control_cac_to_oscal(cac_control, group_id, control_path, parent)
+            # 4b. Find a control to merge into
+            matched_control = None
+            for control in parent.controls:
+                if control.id == new_control.id:
+                    matched_control = control
+                    break
+            # 4c. Merge mapped cac control into oscal control (note: CatalogAPI.merge_catalog doesn't work for this)
+            if not matched_control:
+                parent.controls.append(new_control)
+            else:
+                # 4c1. params
+                if new_control.params:
+                    for new_param in new_control.params:
+                        if matched_control.params:
+                            matched_params = [mp.id for mp in matched_control.params]
+                            if new_param.id not in matched_params:
+                                matched_control.params.append(new_param)
+                # 4c2. props
+                if new_control.props:
+                    for new_prop in new_control.props:
+                        if matched_control.props:
+                            matched_props = [mp.name for mp in matched_control.props]
+                            if new_prop.name not in matched_props:
+                                matched_control.props.append(new_prop)
+                # 4c3. links
+                if new_control.links:
+                    for new_link in new_control.links:
+                        if matched_control.links:
+                            matched_links = [mp.href for mp in matched_control.links]
+                            if new_link.href not in matched_links:
+                                matched_control.links.append(new_link)
+                # 4c4. parts
+                if new_control.parts:
+                    for new_part in new_control.parts:
+                        if matched_control.parts:
+                            matched_parts = [mp.id for mp in matched_control.parts]
+                            if new_part.id not in matched_parts:
+                                matched_control.parts.append(new_part)
+
+    def _create_or_update_catalog(self, policy: Policy) -> None:
+        """Create or update catalog for specified CaC profile."""
+        repo_path = pathlib.Path(self.working_dir)
+        oscal_json = repo_path / "catalogs" / self.oscal_catalog / "catalog.json"
```

Comment:
Nit: trestle has a `function` you could use do you don't have to use a hard-code the path details

https://github.com/oscal-compass/compliance-trestle/blob/729bf4454ee5006915a156e52e6c570c4311b7a4/trestle/common/load_validate.py#L42

Reply:
The function raises an exception if the catalog file doesn't exist, and get_model_path_for_name_and_class also returns None if the file doesn't exist. It doesn't seem like there's a nice way to get the path to a file for future creation